shared/src/main/scala/coop/rchain/casper/lcasper/LevelView.scala

```diff
@@ -0,0 +1,78 @@
+package coop.rchain.casper.lcasper
+import coop.rchain.casper.lcasper.Casper.{Fringe, LevelReconciliation}
+
+import scala.annotation.tailrec
+
+// View on DAG level
+final case class LevelView[M, S](height: Long, partitions: Vector[Fringe[M, S]]) {
+  def fringe: Fringe[M, S] = partitions.reduceOption(_ ++ _).getOrElse(Map())
+  def stake(bondsMap: Map[S, Long]): Long =
+    fringe.keysIterator.map { s =>
+      val bond = bondsMap.get(s)
+      assert(bond.isDefined, "Level view contains message from not bonded sender.")
+      bond.get
+    }.sum
+}
+
+object LevelView {
+
+  def reconcile[M, S](views: Vector[LevelView[M, S]])(
+      bondsMap: Map[S, Long]
+  ): LevelReconciliation[M, S] = {
+    require(
+      views.map(_.height).distinct.size == 1,
+      "Unexpected error: reconciliation across different levels is detected."
+    )
+
+    final case class PastureState(
+        amoeba: Vector[(Fringe[M, S], Int)],          // indexed legs of amoeba
+        pasture: Vector[Vector[(Fringe[M, S], Int)]], // indexed pieces of food in the pasture
```

Comment:
What is the meaning of index? Vector has index of elements and Int is in tuple, which index you are referring to?

Reply:
Input for this algo is list of level views, which are list of non overlapping partitions. 
So when partition Psmall from view V1 is absorbed by Pbig in V2 - we should remove absorbed partition from V1 and replace partition with the aggregate one in V2. To be able to do this we need to be able to remove item with some index from a vector and replace item by its index. So this index is used for this purpose.